### ISSUE TITLE
Panel Icon Size

### ISSUE THREAD

**Issue by in4matix** [Rmano/gse-switch-focus-mode#10]
First, thank you for this extension, It is extremely useful.
I have this issue (I think it has most likely to do with the shell theme that I use) the panel icon is extremely small (hardly visible)
My system is Ubuntu 22.04 LTS (Gnome 42.9) Wayland and I have extension version 11 from the Gnome extensions website.
I have tried the following by editing extension.js with no luck:
```
	icon_f = new St.Icon({ style_class: 'system-status-icon', icon_size: 48 });
	icon_f.gicon = Gio.icon_new_for_string(Me.path + '/icons/fmode-symbolic.svg');
	icon_c = new St.Icon({ style_class: 'system-status-icon', icon_size: 48 });
	icon_c.gicon = Gio.icon_new_for_string(Me.path + '/icons/cmode-symbolic.svg');
```
If you have the time, I would really appreciate if you could assist me.

**Comment by Rmano** [comment]
Hmm... I am not sure what could be happening here.  If I use `just perfection` shell extension to boost the panel size to 48px it looks horrible ;-) but it correctly scales:

![image](https://github.com/user-attachments/assets/a9758a94-d4ea-46a9-a5bc-6bce80b7d511)

the icon is in SVG, so it should render at whichever size you need...

**Comment by in4matix** [comment]
I don't think it has anything to do with panel size. All other extension icons appear correctly. With Tiling Shell I had to put a slight border around it's icon because it was slightly larger than the other icons. Perhaps it has something to do with my icon theme? I use a modified version of Deepin. I really would like this extension icon to match the others :-(
![Screenshot from 2024-09-22 18-46-17](https://github.com/user-attachments/assets/fe3b3ea0-5fb4-4991-97ba-d9cfeb89a4a2)

**Comment by Rmano** [comment]
I am sorry --- I have no idea about what's happening. The extension is bog simple; it does not check nor react to any icon theme... if you want to change the icons, it should suffice to drop your SVGs in `.local/share/gnome-shell/extensions/SwitchFocusType@romano.rgtti.com/icons` instead of the `cmode.svg` and `fmode.svg` ones located there, and restart the shell...

**Comment by in4matix** [comment]
I tried changing the icon, even a 512x512 SVG, no luck. I'm at wits-end too...

**Comment by in4matix** [comment]
Not sure if this is of any help: https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/200

**Comment by Rmano** [comment]
Hmmm. I don't think it matters. Yes, App indicators icons show different and don't scale (look at remmina in my sceenshot) but my extension icons are native to GNOME...

**Comment by in4matix** [comment]
Hi, I installed the extension on a second PC which runs Zorin OS 17.2 Pro (I think it is Gnome 33), and I am having this same issue:
![Screenshot from 2024-09-23 19-26-24](https://github.com/user-attachments/assets/af9b4d72-e05b-4695-9ec7-2f317e2e8949)

**Comment by Rmano** [comment]
So now I am confused. I copied and modified the icon from the app "lock screen", and look:

![image](https://github.com/user-attachments/assets/994e316d-c8d2-488f-94d9-ffc33962f860)

apart from the colors, the icon from my extension is the same size as the others --- or am I missing something more?

**Comment by in4matix** [comment]
I didn't spend any time trying to change anything after installing on the Zorin PC, and looking at it closer now, it looks like the icon just appears to be smaller because it's an overall light icon on a light background in this case...

**Comment by Rmano** [comment]
You can try to open the icon in InkScape and try to invert it?

**Comment by in4matix** [comment]
It's a petty their is no preference setting to choose own icon and size maybe... That would be quite useful to try different icons and sizes without having to constantly meddle with the code...

**Comment by Rmano** [comment]
Yes, I agree, but dealing with preferences is quite a pain. I did something similar in my other extension (https://extensions.gnome.org/extension/6646/touchpad-on-off/), but it's a pain to use things outside the extension directory...

**Comment by in4matix** [comment]
Perhaps it might be a good balance to have the preference setting and a couple of different icon styles in the icons folder. Another option could be to use the Freedesktop Standard Icon naming, where one just types in the icon name instead of having an icons folder (similar to another extension I use: https://github.com/arunk140/gnome-command-menu

**Comment by in4matix** [comment]
Or, I would not mind the ability to use text labels instead of icons...

**Comment by in4matix** [comment]
I am still learning to code on my own as a hobby and if I wasn't so stupid I would help you with this, but after all the research I've done, I still can't grasp even basic coding :-(
